### ISSUE TITLE
Prohibit duality in application bindings

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/lints#248]
In EO we don't support application with and without binding at the same time, so the next code is invalid:
```
man
  "Jeff":name # "Jeff" object is bound to the "name" attribute of "man"
  42          # "42" is bound to 2nd attribute of "man" but not explicitly
```

All application arguments must be either with `@as` attribute or without it:

The next XMIR is valid:
```xml
<o base="man">
  <o base="string"/>
  <o base="number"/>
</o>
```

The next XMIR is also valid:
```xml
<o base="man">
  <o base="string" as="name"/>
  <o base="number" as="age"/>
</o>
```

But the next XMIR is not valid:
```xml
<o base="man">
  <o base="string" as="name"/>
  <o base="number"/>
</o>
```

Let's add check that would catch such cases with severity `critical`

**Comment by yegor256** [comment]
@h1alexbel can you help?

**Comment by h1alexbel** [comment]
@yegor256 yes
